sycl/doc/extensions/proposed/sycl_ext_oneapi_work_group_static.asciidoc

```diff
@@ -289,26 +302,13 @@ This non-normative section provides information about one possible
 implementation of this extension.  It is not part of the specification of the
 extension's API.
 
-For class types and bounded arrays, the class can be implemented on top of
-the existing `__sycl_allocateLocalMemory` intrinsic:
-[source,c++]
-----
-#ifdef __SYCL_DEVICE_ONLY__
-  __attribute__((opencl_local)) T *ptr = reinterpret_cast<__attribute__((opencl_local)) T *>(__sycl_allocateLocalMemory(sizeof(T), alignof(T)));
-#else
-  T *ptr{};
-#endif
-----
-
-Note, however, that implementing the correct semantics may require some
-adjustment to the handling of this intrinsic. A simple class as written above
-would create a separate allocation for every call to an inlined function.
-Creating device local memory allocations should be handled before inlining to
-prevent this.
+For class types and bounded arrays, the class can be implemented using
+the WG scope attribute used by hierarchical implementation.
+Note, however, that this requires to expose the attribute to the users.
```

Comment:
Why would this expose the attribute to the user?  Are you worried that users will read the header files and discover implementation details?  There are all sorts of implementation details in the headers, so why is this special?

Do we still need this "Implementation notes" section?  Maybe it's time to delete it.